Reference/Common-Pitfalls/index.md

```diff
@@ -0,0 +1,466 @@
+your application.
+
+Your views should rely only on the readonly data access of the `UmbracoHelper` and the properties/methods that it exposes. This ensures
+that the data being queried is fast (comes from cache) and that you aren't inadvertently making database changes.
+
+If you are using `Application.Services...` in your views, you should figure out why this is being done and in most cases remove this logic.   
+
+## Using UmbracoContext to access ApplicationContext
+
+You should not access the `ApplicationContext` via the `UmbracoContext`. 
+
+For example: `UmbracoContext.Current.Application` _<-- this is now deprecated/obsolete_
+
+If you need access to both the `UmbracoContext` and the `ApplicationContext` you should either 
+
+* Access these services via the properties exposed on the Umbraco base class you are using (i.e. Controllers, views, controls, http handler, etc...)
+* or inject these services into the services you are using 
+* or access each of these services from their own singleton constructs: `UmbracoContext.Current` and `ApplicationContext.Current`.
+
+## Using Umbraco content items for volatile data 
+
+This is one of the worst Umbraco anti-patterns and could very well cause your site to perform ultra poorly. 
+
+Umbraco's content should not be used for volatile data, Umbraco's APIs and the way Umbraco's data is persisted
+was never designed for this. If you need to store/write/track data that changes a lot you should use a 
+custom database table or another service but not Umbraco content nodes.
+
+Some examples of what not to do are:
+
+* Hit counters to track the number of times your page has been viewed - use something like Google Analytics for this or a custom database table
+* Creating new nodes for form submissions - this should be stored in a custom database table
+* Importing lots of data into Umbraco content nodes that could easily just be stored in a custom database table (i.e. it's not going to be edited).
+In some cases this might be ok but many times we've seen bulk imports occur on a hourly/daily schedule which is generally unecessary.
+
+## Processing during startup
+
+Umbraco allows you to run some initialization code during startup by using `ApplicationEventHandler`, however great
+care should be used to ensure that you are not slowing down application startup. You should be especially careful
+as a Package developer that you are not slowing down application startup since your package may end up being used for
+thousands of websites.
+
+In many cases, [initialization code can be done lazily instead of eagerly](https://msdn.microsoft.com/en-us/library/dd997286(v=vs.110).aspx). 
+Instead of initialization everything you need as soon as the application starts you could execute your initialization code only when it is required. 
+This can be achieved in various ways such as:
+
+* Using [`Lazy<T>`](https://msdn.microsoft.com/en-us/library/dd642331(v=vs.110).aspx) and put the initialization logic in it's callback
```

Comment:
Could a super simple example of how to do this be given. As even Umbraco Forms itself does not follow this pattern fully & initialises some stuff in startup however does leave a marker file to avoid re-checking all the time.